p2p/libp2p/netMessenger.go

```diff
@@ -542,6 +545,21 @@ func (netMes *networkMessenger) ThresholdMinConnectedPeers() int {
 	return netMes.connMonitor.thresholdMinConnectedPeers
 }
 
+// SetPeerShardResolver sets the peer shard resolver component that is able to resolve the link
+// between p2p.PeerID and shardId
+func (netMes *networkMessenger) SetPeerShardResolver(peerShardResolver p2p.PeerShardResolver) error {
+	if check.IfNil(peerShardResolver) {
+		return p2p.ErrNilPeerShardResolver
+	}
+
+	kadSharder, err := networksharding.NewKadSharder(kadSharderPrioBits, peerShardResolver)
```

Comment:
Will think about a solution to this. The whole file is swarming with creating new implementations because of the need to create the sub-components that do the job done.

Reply:
Refactoring the netMessenger struct is too complicated for this PR. Created EN-5014, please check description. Will remain as it is for the moment